apps/prairielearn/python/prairielearn/core.py

```diff
@@ -1820,8 +1820,8 @@ def clean_identifier_name(name: str) -> str:
     """Escapes a string so that it becomes a valid Python identifier."""
     # Strip invalid characters and weird leading characters so we have
     # a decent python identifier
-    name = re.sub("[^a-zA-Z0-9_]", "_", name)
-    name = re.sub("^[^a-zA-Z]+", "", name)
+    name = re.sub(r"[^a-zA-Z0-9_]", "_", name)
```

Comment:
Shouldn't matter but would be nice to have a couple of small test cases for this (even if they're just written with AI)

Reply:
Agreed. After all the recent breakages, I'm _this_ close to mandating comprehensive test coverage for all changes to `core.py` functions.

Reply:
Added some tests in 6714b776e58a71722a1fd62aac13bd1b9437ea36 and bfc4f0d07c622ef75c127162ce8b59fbfd521c13.